### PR TITLE
Fogbugz 8477: Crash on loading older domains that have zones without the haze component.

### PR DESCRIPTION
**Pull request by NissimHadar**
The crash occurred when:
   1) run latest master build in Eschatology or any domain with zones
   2) open tablet
   3) choose "CREATE"
   4) choose "list"
  5) observer deadlock and/or crash

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4287/HighFidelity-Beta-PR11634-e314ee432e9ca5393d0237a09c22fcd1f87f4520.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4292/HighFidelity-Beta-PR11634-d01ecb71dfc607b6e32d87df3c0065fdd89bc63c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4292/HighFidelity-Beta-PR11634-d01ecb71dfc607b6e32d87df3c0065fdd89bc63c.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4298/HighFidelity-Beta-PR11634-0f519da76e4a13d962e07ae77138527185bc64ec.dmg

**Comment by samcake** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4302/HighFidelity-Beta-PR11634-0f519da76e4a13d962e07ae77138527185bc64ec.dmg

**Comment by NissimHadar** [comment]
It builds fine, on all platforms.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4304/HighFidelity-Beta-PR11634-0f519da76e4a13d962e07ae77138527185bc64ec.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4304/HighFidelity-Beta-PR11634-0f519da76e4a13d962e07ae77138527185bc64ec.exe

File: libraries/entities/src/ZoneEntityItem.cpp [hunk 1]
```diff
@@ -321,7 +322,13 @@ void ZoneEntityItem::resetRenderingPropertiesChanged() {
 }
 
 void ZoneEntityItem::setHazeMode(const uint32_t value) {
-    _hazeMode = value;
+    if (value < COMPONENT_MODE_ITEM_COUNT) {
+        _hazeMode = value;
+    }
+    else {
```
Comment (sethalves) [inline libraries/entities/src/ZoneEntityItem.cpp]: else should be on previous line

**Comment by Nex-Pro** [comment]
T E S T I N G win10

**Comment by Nex-Pro** [comment]
@NissimHadar Go to eschatology and check it doesn't crash ?

File: libraries/entities/src/ZoneEntityItem.cpp (hunk 1 above)
Comment (Nex-Pro) [reply libraries/entities/src/ZoneEntityItem.cpp]: else}{

@sethalves  that should do it !   ;)

**Comment by Nex-Pro** [comment]
### PASS

Went to Eschatology in this build.
Opened the entity list.
Checked that there were zones in the list.
Also created and deleted a zone.
I survived !

**Comment by Nex-Pro** [comment]
### No crashes to report ! @NissimHadar @sethalves

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4321/HighFidelity-Beta-PR11634-00c99d3d02dd84322eca21bc01121e9ff0015e78.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4321/HighFidelity-Beta-PR11634-00c99d3d02dd84322eca21bc01121e9ff0015e78.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4325/HighFidelity-Beta-PR11634-5d0deaf9462201848a86f3f8763e0dae20e4cec4.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4325/HighFidelity-Beta-PR11634-5d0deaf9462201848a86f3f8763e0dae20e4cec4.exe

**Comment by samcake** [comment]
Testing again hang on

**Comment by samcake** [comment]
tested again, PASS

**Pull request merged**